common/src/main/java/io/netty/util/internal/ObjectUtil.java

```diff
@@ -35,13 +49,42 @@ private ObjectUtil() {
         return arg;
     }
 
+    /**
+     * Checks that the given argument is not null. If it is, throws {@link IllegalArgumentException}.
+     * Otherwise, returns the argument.
+     */
+    public static <T> T checkNotNullWithIAE(final T arg, final String paramName) throws IllegalArgumentException {
+        if (arg == null) {
+            throw new IllegalArgumentException(String.format(EX_PARAM_MUST_NOT_BE_NULL, paramName));
```

Comment:
I'm personally not a fan of using `String.format(...)` for exceptions (or logging) because it itself can throw an exception if any of the arguments are incorrect. You basically need explicit Unit Tests for that stuff too to be sure it's behaving as expected.

Reply:
Thanks, I will add some.

Reply:
Good point @rkapsi

Reply:
@rkapsi A simple assert("message expected", actualMessage) fails - I18N, working on a local machine with German settings compared to the build server with English settings. Maybe it's easier to turn to String concatenation.